File: src/app/modules/m1-school-office/data-access/models/training-course-offering.ts
export interface TrainingCourseOffering {
  id: number;
  directorateId: number | null;
  schoolId: number | null;
  courseCode: string;
  courseTitleAr: string;
  trainerName: string | null;
  startDate: string;
  endDate: string;
  totalHours: number;
  maxParticipants: number;
  costPerParticipant: number;
  courseTitleEn: string | null;
  trainingLocation: string | null;
  targetSpecialization: string | null;
  enrolledParticipantsCount: number;
  certificateTemplateUrl: string | null;
  isActive: boolean;
}

export type CreateTrainingCourseOfferingDto = Omit<TrainingCourseOffering, 'id' | 'isActive'>;

export type UpdateTrainingCourseOfferingDto = Omit<TrainingCourseOffering, 'directorateId' | 'schoolId' | 'isActive'>;

File: 01_Backend/EduMS.Backend/src/EduMS.Domain/Entities/M1_SchoolAdmin/TrainingCourseOffering.cs
using EduMS.Domain.Common;

namespace EduMS.Domain.Entities;

public class TrainingCourseOffering : BaseAuditableEntity
{
    public long? DirectorateId { get; set; }
    public long? SchoolId { get; set; }
    public string CourseCode { get; set; } = string.Empty;
    public string CourseTitleAr { get; set; } = string.Empty;
    public string? TrainerName { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public int TotalHours { get; set; }
    public int MaxParticipants { get; set; }
    public decimal CostPerParticipant { get; set; }
    public string? CourseTitleEn { get; set; }
    public string? TrainingLocation { get; set; }
    public string? TargetSpecialization { get; set; }
    public int EnrolledParticipantsCount { get; set; }
    public string? CertificateTemplateUrl { get; set; }
    public bool IsActive { get; set; } = true;

    // Navigation Properties
    public virtual Directorate? Directorate { get; set; }
    public virtual School? School { get; set; }
}
